source/Calamari.AzureAppService.Tests/Legacy/LegacyAppServiceIntegrationTest.cs

```diff
@@ -103,12 +101,19 @@ public async Task Cleanup()
 
         protected async Task AssertContent(string hostName, string actualText, string rootPath = null)
         {
-            var response = await RetryPolicies.TransientHttpErrorsPolicy.ExecuteAsync(async () =>
-                                                                                      {
-                                                                                          var r = await client.GetAsync($"https://{hostName}/{rootPath}");
-                                                                                          r.EnsureSuccessStatusCode();
-                                                                                          return r;
-                                                                                      });
+            var response = await RetryPolicies.TestsTransientHttpErrorsPolicy.ExecuteAsync(async context =>
+                                                                                           {
+                                                                                               var r = await client.GetAsync($"https://{hostName}/{rootPath}");
+                                                                                               if (!r.IsSuccessStatusCode)
+                                                                                               {
```

Comment:
The deploy a war package still appears to be flakey, although I think less so. On test failures this returns a 500, theres a SO thread on setting the AppService to AlwaysOn but that already seems to be the case for these resources.